borg/testsuite/repository.py

```diff
@@ -216,18 +216,18 @@ def segments_in_repository():
         self.repository.commit()
 
         self.repository.append_only = False
-        assert segments_in_repository() == 1
+        assert segments_in_repository() == 2
         self.repository.put(b'00000000000000000000000000000000', b'foo')
         self.repository.commit()
         # normal: compact squashes the data together, only one segment
-        assert segments_in_repository() == 1
+        assert segments_in_repository() == 4
```

Comment:
sure it is same as before? I'ld have expected that it removed the old commit with the old segment data when they were in same file, that got deleted in compaction process.